internal/identity/identitymanager.go

```diff
@@ -0,0 +1,242 @@
+	if di == nil || ii == nil || bi == nil {
+		return nil, i18n.NewError(ctx, i18n.MsgInitializationNilDepError)
+	}
+	im := &identityManager{
+		database:           di,
+		plugin:             ii,
+		blockchain:         bi,
+		identityCacheTTL:   config.GetDuration(config.IdentityManagerCacheTTL),
+		signingKeyCacheTTL: config.GetDuration(config.IdentityManagerCacheTTL),
+	}
+	// For the identity and signingkey caches, we just treat them all equally sized and the max items
+	im.identityCache = ccache.New(
+		ccache.Configure().MaxSize(config.GetInt64(config.IdentityManagerCacheLimit)),
+	)
+	im.signingKeyCache = ccache.New(
+		ccache.Configure().MaxSize(config.GetInt64(config.IdentityManagerCacheLimit)),
+	)
+
+	return im, nil
+}
+
+func (im *identityManager) GetLocalOrganization(ctx context.Context) (*fftypes.Organization, error) {
+	return im.cachedOrgLookupByAuthor(ctx, im.localOrgDID)
+}
+
+func (im *identityManager) OrgDID(org *fftypes.Organization) string {
+	return org.GetDID()
+}
+
+// ResolveInputIdentity takes in identity input information from an API call, or configuration load, and resolves
+// the combination
+func (im *identityManager) ResolveInputIdentity(ctx context.Context, identity *fftypes.Identity) (err error) {
+	log.L(ctx).Debugf("Resolving identity input: key='%s' author='%s'", identity.Key, identity.Author)
+
+	identity.Key, err = im.ResolveSigningKey(ctx, identity.Key)
+	if err != nil {
+		return err
+	}
+
+	// Resolve the identity
+	if err = im.resolveInputAuthor(ctx, identity); err != nil {
+		return err
+	}
+
+	log.L(ctx).Debugf("Resolved identity: key='%s' author='%s'", identity.Key, identity.Author)
+	return
+}
+
+func (im *identityManager) ResolveSigningKeyIdentity(ctx context.Context, signingKey string) (author string, err error) {
+
+	signingKey, err = im.ResolveSigningKey(ctx, signingKey)
+	if err != nil {
+		return "", err
+	}
+
+	// TODO: Consider other ways identity could be resolved
+	org, err := im.cachedOrgLookupBySigningKey(ctx, signingKey)
+	if err != nil {
+		return "", err
+	}
+
+	return im.OrgDID(org), nil
+
+}
+
+func (im *identityManager) ResolveLocalOrgDID(ctx context.Context) (localOrgDID string, err error) {
+	if im.localOrgDID != "" {
+		return im.localOrgDID, nil
+	}
+	orgKey := config.GetString(config.OrgKey)
+	if orgKey == "" {
+		orgKey = config.GetString(config.OrgIdentityDeprecated)
+	}
+	im.localOrgDID, err = im.ResolveSigningKeyIdentity(ctx, orgKey)
+	if err != nil {
+		return "", i18n.WrapError(ctx, err, i18n.MsgLocalOrgLookupFailed, orgKey)
+	}
+	if im.localOrgDID == "" {
+		return "", i18n.NewError(ctx, i18n.MsgLocalOrgLookupFailed, orgKey)
+	}
+	return im.localOrgDID, err
+}
+
+func (im *identityManager) ResolveSigningKey(ctx context.Context, inputKey string) (outputKey string, err error) {
+	// Resolve the signing key
+	if inputKey != "" {
+		if cached := im.signingKeyCache.Get(inputKey); cached != nil {
+			cached.Extend(im.identityCacheTTL)
```

Comment:
Is it necessary to manually `Extend` each time we fetch? Shouldn't this be handled transparently by the LRU cache?

Reply:
I didn't find clear docs on this (the code you describe is elsewhere in FireFly).

Digging into the code here...
https://github.com/karlseguin/ccache/blob/ef4bd54683f611ad3df9ba84689234c78243c464/cache.go#L101-L117

Then here...
https://github.com/karlseguin/ccache/blob/ef4bd54683f611ad3df9ba84689234c78243c464/cache.go#L394-L410

It seems the behavior if you _don't_ extend, is that it will end up in the purge list, for LRU deletion.
While `Get` alone will make it less likely to be the thing that's deleted, that's not the same as ensuring it's retained for a period after being explicitly accessed.

I guess my vote is to leave the pattern as it is for the moment in this PR, then consider a wrapper around the caching framework in a separate PR that handles optimization of the logic.